tests/test_cli_backend.py

```
@@ -303,6 +307,196 @@ def fake_run(cmd, **kw):
+    assert cb.os.path.isabs(result)                      # 必须绝对
+    assert result == cb.os.path.abspath("./bin/codex")
+
+
+def test_run_codex_execs_resolved_abspath(monkeypatch):
+    """_run_codex 用解析出的绝对路径当 argv[0]（修 GUI 启动找不到 codex）。"""
+    cb._BIN_CACHE.clear()
+    monkeypatch.setattr(cb, "_resolve_cli_bin", lambda name, configured: "/Users/x/.local/bin/codex")
+    captured = {}
+
+    class _P:
+        stdout = "ok"; stderr = ""; returncode = 0
```

Ruff (0.15.15)

[error] Multiple statements on one line (semicolon)

(E702)